packages/flutter/lib/src/rendering/sliver_fill.dart

```diff
@@ -59,69 +59,90 @@ class RenderSliverFillViewport extends RenderSliverFixedExtentBoxAdaptor {
-  /// [precedingScrollExtent] exceeds the size of the viewport, the sliver will
-  /// defer to the child's size rather than overriding it.
-  bool hasScrollBody;
-
-  /// Indicates whether the child should stretch to fill the overscroll area
-  /// created by certain scroll physics, such as iOS' default scroll physics.
-  /// This value cannot be null. This flag is only relevant when the
-  /// [hasScrollBody] value is false.
-  ///
-  /// Defaults to false, meaning the default behavior is for the child to
-  /// maintain its size and not extend into the overscroll area.
-  bool fillOverscroll;
+class RenderSliverFillRemainingWithScrollable extends RenderSliverSingleBoxAdapter {
+  /// Creates a [RenderSliver] that wraps a scrollable [RenderBox] which is
+  /// sized to fit the remaining space in the viewport.
+  RenderSliverFillRemainingWithScrollable({ RenderBox child }) : super(child: child);
 
   @override
   void performLayout() {
-    double childExtent;
+    // TODO(Piinks): This may fill too much space for NestedScrollView, https://github.com/flutter/flutter/issues/46028
+    final double extent = constraints.remainingPaintExtent - math.min(constraints.overlap, 0.0);
+
+    if (child != null)
+      child.layout(constraints.asBoxConstraints(
+        minExtent: extent,
+        maxExtent: extent,
+      ));
+
+    final double paintedChildSize = calculatePaintOffset(constraints, from: 0.0, to: extent);
+    assert(paintedChildSize.isFinite);
+    assert(paintedChildSize >= 0.0);
+    geometry = SliverGeometry(
+      scrollExtent: constraints.viewportMainAxisExtent,
+      paintExtent: paintedChildSize,
+      maxPaintExtent: paintedChildSize,
+      hasVisualOverflow: extent > constraints.remainingPaintExtent || constraints.scrollOffset > 0.0,
+    );
+    if (child != null)
+      setChildParentData(child, constraints, geometry);
+  }
+}
+
+/// A sliver that contains a single box child that is non-scrollable and fills
+/// the remaining space in the viewport.
+///
+/// [RenderSliverFillRemainingWithoutScrollable] sizes its child to fill the
+/// viewport in the cross axis and to fill the remaining space in the viewport
+/// in the main axis.
+///
+/// Typically this will be the last sliver in a viewport, since (by definition)
+/// there is never any room for anything beyond this sliver.
+///
+/// See also:
+///
+///  * [RenderSliverFillRemainingWithScrollable], which lays out its scrollable
+///    child slightly different than this widget.
+///  * [RenderSliverFillRemainingAndOverscroll], which incorporates the
+///    overscroll into the remaining space to fill.
+///  * [RenderSliverFillViewport], which sizes its children based on the
+///    size of the viewport, regardless of what else is in the scroll view.
+///  * [RenderSliverList], which shows a list of variable-sized children in a
+///    viewport.
+class RenderSliverFillRemainingWithoutScrollable extends RenderSliverSingleBoxAdapter {
```

Comment:
"WithoutScrollable" is a strange name. Can we name this in terms of what it does and not what it doesn't do?

Reply:
Maybe this should just be "RenderSliverFilRemaining" since it kinda represents the default case?